qrgui/mainWindow/scriptAPI/utils.h

```diff
@@ -0,0 +1,128 @@
+/* Copyright 2015 QReal Research Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. */
+
+#pragma once
+
+#include <QtWidgets/QMenu>
+#include <QtWidgets/QAction>
+#include <functional>
+
+namespace qReal {
+
+class MainWindow;
+
+namespace gui {
+
+class GuiFacade;
+class VirtualCursor;
+class VirtualKeyboard;
+class SceneAPI;
+class PaletteAPI;
+class HintAPI;
+class ScriptAPI;
+
+/// This class provides scripting access to some complicated actions
+/// and usefull methods for test writing, independent of ui, scene, palette etc.
+class Utils : public QObject
+{
+	Q_OBJECT
+
+public:
+	Utils(ScriptAPI &scriptAPI, MainWindow &mainWindow, VirtualCursor &virtualCursor, HintAPI &hintAPI);
+
+	// Actions: Don't forget about virtual cursor's moves.
+	/// @note This method works with a keyboard (not a mouse).
+	/// Activates menu getting with pointer.
+	// todo: realise visible mouse moves with hints or virtual cursor.
+	Q_INVOKABLE void activateMenu(QMenu *menu) noexcept;
+
+	/// @warning Use this method only after opening of the assigned menu.
+	/// @note This method works with a keyboard (not a mouse).
```

Comment:
This method interacts with QMenu through the virtual keyboard, not keyboard mouse